src/main/kotlin/sliv/tool/parsers/CSVParser.kt

```diff
@@ -0,0 +1,29 @@
+package sliv.tool.parsers
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.csv.CsvMapper
+import com.fasterxml.jackson.dataformat.csv.CsvSchema
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+
+// A base class for CSV parsers.
+abstract class CSVParser<T>(private val format: Class<T>) : Parser<T> {
+    protected val mapper: ObjectMapper = CsvMapper().registerModule(KotlinModule.Builder().build())
+
+    override fun parse(filePath: String): List<T> {
+        val textData = ParserUtils.readFileText(filePath) ?: return emptyList()
+
+        val csvSchema = mapper.readerFor(format).with(CsvSchema.emptySchema().withHeader())
+        return csvSchema.readValues<T>(textData).readAll()
+    }
+
+    override fun extractUIDs(filePath: String): List<Long> {
+        val textData = ParserUtils.readFileText(filePath) ?: return emptyList()
+
+        val uids = mutableListOf<Long>()
+        val csvSchema = mapper.readerFor(Map::class.java).with(CsvSchema.emptySchema().withHeader())
+        val mappingIterator = csvSchema.readValues<Map<String, Any>>(textData)
+        mappingIterator.forEach { uids.add((it["uid"] as String).toLong()) }
```

Comment:
can be `map` applied here?

Reply:
why we cant call previous method and just map to uids?

Reply:
I thought that this would require extra actions and decided to find a way to extract only one field. Also I measured the running time on a test file with a large number of lines and this method was around 1.5 times faster